src/stacked_borrows/stack.rs

```diff
@@ -0,0 +1,258 @@
+use crate::stacked_borrows::{AccessKind, Item, Permission, SbTag};
+use std::ops::Range;
+
+const CACHE_LEN: usize = 32;
```

Comment:
Please add a comment giving at least some inkling of how this number was chosen. I assume there's a tradeoff here, bigger cache is slower to look up in?